fdbbackup/FileDecoder.actor.cpp

```diff
@@ -814,6 +830,34 @@ ACTOR Future<Void> process_file(Reference<IBackupContainer> container,
 	return Void();
 }
 
+// Use the snapshot metadata to quickly identify relevant range files and
+// then filter by versions.
+ACTOR Future<std::vector<RangeFile>> getRangeFiles(Reference<IBackupContainer> bc, Reference<DecodeParams> params) {
+	state std::vector<KeyspaceSnapshotFile> snapshots =
+	    wait((dynamic_cast<BackupContainerFileSystem*>(bc.getPtr()))->listKeyspaceSnapshots());
+	state std::vector<RangeFile> files;
+
+	state int i = 0;
+	for (; i < snapshots.size(); i++) {
+		try {
+			std::pair<std::vector<RangeFile>, std::map<std::string, KeyRange>> results =
+			    wait((dynamic_cast<BackupContainerFileSystem*>(bc.getPtr()))->readKeyspaceSnapshot(snapshots[i]));
+			for (const auto& rangeFile : results.first) {
+				const auto& keyRange = results.second.at(rangeFile.fileName);
+				if (params->matchFilters(keyRange)) {
+					files.push_back(rangeFile);
+				}
+			}
```

Comment:
Break from the loop when there is no error?

Reply:
No. If there are two snapshots within the version range, we should gather all files for both.

Reply:
How will the range files be used? Say there are multiple range files for the same key within the interested beginVersion and endVersion?

Reply:
It's up to the user to decide that. This tool just output all matching `(version, key, value)` tuples.